### PR TITLE
Weighted loss

### PR DESCRIPTION
**Pull request by mariaogryczak**
The weighted loss was added.
Weights are saved in the same dict, where classes for heads in the from: "head_name": [number_of_classes, weight_for_head_loss].
Sample weights are provided.

### CODE REVIEW

File: utils/classification_mode.py [hunk 1]
```diff
@@ -3,22 +3,22 @@
 
 def create_heads_dict(classification_mode):
     if classification_mode == "gloss":
-        num_classes_dict = {"gloss": 2400}  # number of classes for each head
-    elif classification_mode == "hamnosys":
+        num_classes_dict = {"gloss": [2400, 1]}  # number of classes for each head
+    elif classification_mode == "hamnosys":  # dla annotacji toy_hamnosys.txt
```
Comment (wiktorfilipiuk) [inline utils/classification_mode.py:7]: let's make this comment english or remove if obsolete :)

File: utils/summary_loss.py [hunk 2]
```diff
@@ -4,19 +4,20 @@
...
 
     def forward(self, predictions: torch.Tensor, targets: torch.Tensor) -> torch.Tensor:
         losses = []
         loss_sum = 0
-        for prediction, target in zip(predictions, targets):
+        for prediction, target, loss_weight in zip(predictions, targets, self.loss_weights):
             one_loss = self.loss(prediction.to("cpu"), target.to("cpu"))
             losses.append(one_loss)
-            loss_sum = loss_sum + one_loss
+            loss_sum = loss_sum + (loss_weight*one_loss)
```
Comment (wiktorfilipiuk) [inline utils/summary_loss.py:22]: Shall we divide the result by sum of all weights?

File: models/model_for_pretraining.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    )
+
+
+class PreTrainingModel(pl.LightningModule):
+    """Awesome model for Gloss Translation"""
+
+    def __init__(
+        self,
+        MODEL_CONFIG = {"lr": 1e-5,
+        "multiply_lr_step": 0.7,
```
Comment (wiktorfilipiuk) [inline models/model_for_pretraining.py:33]: Can we improve indentation, i.e. MODEL_CONFIG = {
    "lr": 1e-5,
    "multiply_lr_step": 0.7,
}

File: utils/summary_loss.py (hunk 2 above)
Comment (AgaMiko) [reply utils/summary_loss.py:22]: we have weights for that

**Review by AgaMiko — APPROVED** [review]

File: utils/classification_mode.py [hunk 4]
```diff
@@ -3,22 +3,23 @@
...
-            "hand_location_x": 5,
-            "hand_location_y": 37,
+            "symmetry_operator": [9, 0],
+            "hand_shape_base_form": [12, 1],
+            "hand_shape_thumb_position": [4, 0],
+            "hand_shape_bending": [6, 0],
+            "hand_position_finger_direction": [18, 0],
+            "hand_position_palm_orientation": [8, 0],
+            "hand_location_x": [5, 0] ,
+            "hand_location_y": [37, 0],
```
Comment (AgaMiko) [inline utils/classification_mode.py:17]: Looks better now! One more thing to consider. Maybe it would be more clear with the dict structure:
```
...
"hand_location_x": {
            "num_class": 5,
            "loss_weight": 0,
            },
"hand_location_y": {
            "num_class": 37,
            "loss_weight": 0,
            },
```

File: models/model.py [hunk 5]
```diff
@@ -57,14 +57,21 @@ def __init__(
             self.run = None
 
         # parameters
+
         self.lr = model_config["lr"]
         self.model_save_dir = model_config["model_save_dir"]
         self.warmup_steps = model_config["warmup_steps"]
         self.multiply_lr_step = model_config["multiply_lr_step"]
         self.num_classes_dict = create_heads_dict(model_config["classification_mode"])
+        print(self.num_classes_dict)
```
Comment (wiktorfilipiuk) [inline models/model.py:66]: remove print

File: models/model_for_pretraining.py [hunk 6]
```diff
@@ -57,36 +56,37 @@ def __init__(
...
+ 
         # parameters
         self.lr = model_config["lr"]
         self.model_save_dir = model_config["model_save_dir"]
         self.warmup_steps = model_config["warmup_steps"]
         self.multiply_lr_step = model_config["multiply_lr_step"]
         self.num_classes_dict = create_heads_dict(model_config["classification_mode"])
+        self.cls_head = []
+        self.loss_weights = []
+        print(self.num_classes_dict)
```
Comment (wiktorfilipiuk) [inline models/model_for_pretraining.py:68]: remove print

**Review by wiktorfilipiuk — APPROVED** [review]
